accepted/2020/feature-switch.md

```diff
@@ -0,0 +1,170 @@
+```xml
+<ItemGroup>
+    ...
+    <RuntimeHostConfigurationOption Include="System.Runtime.OptionalFeatureBehavior"
+                                    Condition="'$(OptionalUserFacingBehavior)' != ''"
+                                    Value="$(OptionalUserFacingBehavior)" />
+    ...
+</ItemGroup>
+```
+
+The name of the property should be picked so that it's clear what `true`/`false` mean. In this case, the idea is that `false` means the feature will be disabled. Other cases (for example `InvariantGlobalization`) might have the opposite polarity.
+
+### Generate the right input for the linker in SDK
+
+All names/values from `RuntimeHostConfigurationOptions` will be passed to the ILLink task, which will apply any feature substitutions defined in [`substitutions.xml`](https://github.com/mono/linker/blob/master/src/linker/README.md#using-custom-substitutions). The substitutions file format will be extended to condition the substitutions based on the feature name/value. Any `RuntimeHostConfigurationOptions` which do not have feature implementations in `substitutions.xml` will not result in any modifications to the IL.
+
+Example of a feature implementation:
+
+```xml
+<linker>
+  <assembly fullname="System.Runtime.FeatureDefiningAssembly" feature="System.Runtime.OptionalFeatureBehavior" featurevalue="false">
+    <type fullname="System.Runtime.FeatureDefiningType">
+      <method signature="System.Boolean get_IsOptionalFeatureEnabled()" body="stub" value="false">
+      </method>
+    </type>
+  </assembly>
+</linker>
+```
+
+Example of setting a feature property from MSBuild.
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net5.0</TargetFramework>
+    <!-- disable the OptionalBehavior feature -->
+    <OptionalBehavior>false</OptionalBehavior>
+  </PropertyGroup>
+</Project>
+```
+
+Similarly, SDK components could set default values for feature properties:
+```xml
+<Project>
+  <!-- build logic for an SDK component -->
+...
+  <PropertyGroup>
+    <OptionalBehavior Condition="'$(OptionalBehavior)' == ''">false</OptionalBehavior>
+  </PropertyGroup>
+...
+</Project>
+```
+SDK authors as always need to ensure that logic is imported in the correct order to allow users to override these default values, if allowed.
+
+## Open questions
+
+### Boolean only or more complex values
+Should the feature switches always represent simple on/off statements and thus be represented as boolean properties? In some cases the simple boolean may necessitate presence of multiple similar feature switches. For example the security algorithms case described above. In the extreme each such algorithm would have its own feature switch - this would obviously create confusion. We could potentially allow feature switches to have string values from a predefined set to solve this problem.
+
+### Which components own the mapping between different representations of feature switches
```

Comment:
I believe this "open question" is now answered above, and this section can be removed now.  (or kept, but marked clearly as "alternate design considerations")

Reply:
Should be fixed now. I also moved one other "open question".